### DISCUSSION TITLE
export of supercharging sessions

### DISCUSSION THREAD

**Discussion by gilrim** [teslamate-org/teslamate#2621] [Q&A] [answered]
I'm currently involved in a lawsuit against Tesla regarding the supercharging throttling of 85's and similar packs in Norway and have been asked to provide exports of the charging sessions over time. I've looked and guess the closest to something usable by the court would be a printout of the different sessions. I'd just need to get the dates added to the detail view as the time is only shown when hovering the graphs. Could this be done easily? I'm not very well versed with grafana so just finding the right data-format came as a challenge :P

The very best would obviously be to have all supercharger sessions below each other on one page with headers for each session as I could then just print one pdf but doing them one-by-one and stitching them up to one after is also doable.

**Comment by rc-16** [comment, marked as answer]
I don't know if this is what you want. I opened the Charges dashboard in Grafana. Next to the word Charges in the middle of the screen is a down arrow. When I clicked it is gave me some options, one was to Inspect. I clicked that and then chose Data. A panel opened with the data and I could then click Download CSV. This allows me to open the data in Excel or Google sheets so that I could manipulate further. 
Hope this helps

**Reply by gilrim** [reply]
Thanks, I've used this method to export the individual charging sessions to csv by opening them from the charges overview and looking at the graphs from them. A bit fiddely to do it for each and every session, but seems like the best available method.

**Reply by ghost** [reply]
Thanks for the answer, it works for my needs of occasional export, but the start/end dates are exported as time series (e.g. start_date_ts) and being a noob I can’t manage to get them in a readable format (the same format I can see them in the “charges” panel). Any idea on how to modify this?

**Comment by cwanja** [comment]
Can you just print to PDF each page and then combine the individual PDFs into a single PDF? The date is at the top of the page.

**Reply by gilrim** [reply]
> Can you just print to PDF each page and then combine the individual PDFs into a single PDF? The date is at the top of the page.

I've tried that, using both firefox and chrome, and for some reason the date picker is minimized when printed:
[Charge Details - Grafana.pdf](https://github.com/adriankumpf/teslamate/files/8608689/Charge.Details.-.Grafana.pdf)
